.circleci/config.yml

```diff
@@ -0,0 +1,85 @@
+version: 2
+jobs:
+  lint:
+    docker:
+      - image: circleci/golang:1.14
+    steps:
+      - checkout
+      - run: make lint
+
+  test:
+    docker:
+      - image: circleci/golang:1.14
+    steps:
+      - checkout
+      - run: make test-and-coverage
+
+  build:
+    docker:
+      - image: circleci/golang:1.14
+    steps:
+      - checkout
+      - run: make build SLO_EXPORTER_VERSION=${CIRCLE_TAG:-$CIRCLE_SHA1}
+      - persist_to_workspace:
+          root: .
+          paths:
+            - build
+
+  docker_release:
+    docker:
+      - image: circleci/golang:1.14
+    environment:
+      DOCKERHUB_USERNAME=sklikdevops
+    steps:
+      - checkout
+      - setup_remote_docker:
+          version: 17.07.0-ce
+      - attach_workspace:
+          at: .
+      - run:
+          command: |
+            cp build/linux-amd64/slo_exporter ./slo_exporter
+            echo "${DOCKERHUB_PASSWORD}" | docker login -u="${DOCKERHUB_USERNAME}" --password-stdin
+            export SLO_EXPORTER_VERSION=${CIRCLE_TAG:-$CIRCLE_SHA1}
+            make docker-build
+            make docker-push
+
+  github_release:
+    docker:
+      - image: circleci/golang:1.14
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          command: |
+            export SLO_EXPORTER_VERSION=${CIRCLE_TAG:-$CIRCLE_SHA1}
+            go get github.com/github-release/github-release
+            make github-release
+
+workflows:
+  version: 2
+  slo-exporter:
+    jobs:
+      - lint
+      - test
+      - build:
+          filters:
+            tags:
+              only: /^v[0-9]+(\.[0-9]+){2}(-.+|[^-.]*)$/
+      - docker_release:
```

Comment:
Maybe we should have separate job for only docker build and run if for every commit and just this release job run for tags?

This way we have no way to verify if the Docker build passes in CI before release.

Reply:
Done within https://github.com/david-vavra/slo-exporter/pull/1/commits/9187899ad35e5543554bec397740f49c2d307ffd